tests/llama_index/test_llama_index_model_export.py

```diff
@@ -0,0 +1,259 @@
+
+
+@pytest.mark.parametrize(
+    "index_fixture",
+    [
+        "single_index",
+        "multi_index",
+        "single_graph",
+    ],
+)
+def test_llama_index_native_log_and_load_model(request, index_fixture):
+    index = request.getfixturevalue(index_fixture)
+    with mlflow.start_run():
+        logged_model = mlflow.llama_index.log_model(index, "model", engine_type="query")
+
+    loaded_model = mlflow.llama_index.load_model(logged_model.model_uri)
+
+    assert "llama_index" in logged_model.flavors
+    assert type(loaded_model) == type(index)
+    engine = loaded_model.as_query_engine()
+    assert engine is not None
+    assert engine.query("Spell llamaindex").response != ""
+
+
+@pytest.mark.parametrize(
+    "engine_type",
+    ["query", "retriever"],
+)
+def test_format_predict_input_correct(single_index, engine_type):
+    wrapped_model = create_engine_wrapper(single_index, engine_type)
+
+    assert isinstance(
+        wrapped_model._format_predict_input(pd.DataFrame({"query_str": ["hi"]})), QueryBundle
+    )
+    assert isinstance(wrapped_model._format_predict_input(np.array(["hi"])), QueryBundle)
+    assert isinstance(wrapped_model._format_predict_input({"query_str": ["hi"]}), QueryBundle)
+    assert isinstance(wrapped_model._format_predict_input({"query_str": "hi"}), QueryBundle)
+    assert isinstance(wrapped_model._format_predict_input(["hi"]), QueryBundle)
+    assert isinstance(wrapped_model._format_predict_input("hi"), QueryBundle)
+
+
+@pytest.mark.parametrize(
+    "engine_type",
+    ["query", "retriever"],
+)
+def test_format_predict_input_incorrect_schema(single_index, engine_type):
+    wrapped_model = create_engine_wrapper(single_index, engine_type)
+
+    with pytest.raises(ValueError, match="correct schema"):
+        wrapped_model._format_predict_input(pd.DataFrame({"incorrect": ["hi"]}))
+    with pytest.raises(ValueError, match="correct schema"):
+        wrapped_model._format_predict_input({"incorrect": ["hi"]})
+
+
+@pytest.mark.parametrize(
+    "engine_type",
+    ["query", "retriever"],
+)
+def test_format_predict_input_correct_schema_complex(single_index, engine_type):
+    wrapped_model = create_engine_wrapper(single_index, engine_type)
+
+    payload = {
+        "query_str": "hi",
+        "image_path": "some/path",
+        "custom_embedding_strs": [["a"]],
+        "embedding": [[1.0]],
+    }
+    assert isinstance(wrapped_model._format_predict_input(pd.DataFrame(payload)), QueryBundle)
+    payload.update(
+        {
+            "custom_embedding_strs": ["a"],
+            "embedding": [1.0],
+        }
+    )
+    assert isinstance(wrapped_model._format_predict_input(payload), QueryBundle)
+
+
+@pytest.mark.parametrize("with_signature", [True, False])
+def test_query_engine_str(tmp_path, single_index, with_signature):
+    payload = "string"
+
+    signature = infer_signature(model_input=payload) if with_signature else None
```

Comment:
Can we pass `input_example` instead of `signature`? MLflow will automatically infer both input and output signature (and we want to validate that inference won't fail, too).